File: freshcart-android/app/src/main/java/com/freshcart/model/Product.kt
package com.freshcart.model

enum class Category(val label: String) {
    FRUITS("Fruits"),
    VEGETABLES("Vegetables"),
}

/** One catalog entry. Prices are whole rupees — the catalog has no paise anywhere. */
data class Product(
    val id: String,
    val emoji: String,
    val name: String,
    val category: Category,
    val weight: String,
    val price: Int,
    val mrp: Int,
) {
    val discounted: Boolean get() = mrp > price
}

data class CartItem(
    val product: Product,
    val quantity: Int,
) {
    val lineTotal: Int get() = product.price * quantity
    val lineSaved: Int get() = (product.mrp - product.price) * quantity
}

fun formatRupees(amount: Int): String = "₹$amount"

File: freshcart-ios/FreshCart/Model/Product.swift
import Foundation

enum ProductCategory: String, CaseIterable {
    case fruits = "Fruits"
    case vegetables = "Vegetables"
}

/// One catalog entry. Prices are whole rupees — the catalog has no paise anywhere,
/// so integer math keeps totals exact.
struct Product: Identifiable, Equatable {
    let id: String
    let emoji: String
    let name: String
    let category: ProductCategory
    let weight: String
    let price: Int
    let mrp: Int

    var isDiscounted: Bool { mrp > price }
}

/// A product plus how many of it sits in the cart.
struct CartItem: Identifiable, Equatable {
    let product: Product
    var quantity: Int

    var id: String { product.id }
    var lineTotal: Int { product.price * quantity }
    var lineMrpTotal: Int { product.mrp * quantity }
}
